tests/unit/modules/network/vyos/test_vyos_file.py

```
@@ -0,0 +1,506 @@
+        # raise an unhandled FileNotFoundError instead of a clean module
+        # error — and this happens even under check_mode, since content
+        # hashing runs before the check-mode short-circuit.
+        result = self._run(
+            {"dest": "/x", "src": "/definitely/does/not/exist/x.pem"},
+            expect_fail=True,
+        )
+        self.assertIn("src not found", result["msg"])
+        self.assertEqual(self.run_commands.call_count, 0)
+
+    def test_rejects_src_that_is_a_directory(self):
+        result = self._run({"dest": "/x", "src": "/tmp"}, expect_fail=True)
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)